### ISSUE TITLE
[Bug]: Search Not Working

### ISSUE THREAD

**Issue by jonchen727** [massCodeIO/massCode#38]
### Describe the bug

Search does not after upgrading to 2.1.0. 

### To reproduce

1. Enter anything into search

### Version

2.1.0

### OS

macOS

### Validations

- [X] Follow our [Code of Conduct](https://github.com/massCodeIO/massCode/blob/master/CODE_OF_CONDUCT.md)
- [X] Check that there isn't [already an issue](https://github.com/massCodeIO/massCode/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/massCodeIO/massCode/discussions).

**Comment by heliomcp** [comment]
Related with my issue closed (https://github.com/massCodeIO/massCode/issues/35).

**Comment by antonreshetov** [comment]
Is there any sequence of actions leading to this bug? Because I can't reproduce this bug.

**Comment by lt-laurap** [comment]
Same problem here.  
It's easier to notice when you try to search within a Folder, you will be redirected to the All Snippets section.

**Comment by antonreshetov** [comment]
@lt-laurap redirecting to All Snippets during a search is normal behavior, because the search is performed on all snippets, not on a selected folder.

**Comment by antonreshetov** [comment]
@jonchen727 Still can't reproduce the bug. Maybe the `db.json` is corrupted?

Search is a very simple function: https://github.com/massCodeIO/massCode/blob/c486fbf8028fa0784009598e182b885c38aab86e/src/renderer/store/snippets.ts#L271

**Comment by jonchen727** [comment]
@antonreshetov 
I found the issue, there was one snippet with a fragment with value null and the other fragment had a value. Now search works as intended, however there is another issue I saw. 

When doing a search and the first fragment does not contain the search term, while the others do there, a weird highlighting occurs when you go into the fragment that contains the search term then to a different snippet, then back to the default fragment. 

<img width="815" alt="image" src="https://user-images.githubusercontent.com/29684743/164036625-4ef8f5b4-19fd-4e8e-899b-27239ca8f90b.png">
